framework/src/base/FEProblemBase.C

```diff
@@ -129,6 +129,9 @@ validParams<FEProblemBase>()
                         false,
                         "This causes PETSc to error if it had to reallocate memory in the Jacobian "
                         "matrix due to not having enough nonzeros");
+  params.addParam<bool>("igore_zeros_jacobian",
```

Comment:
`ignore_jacobian_zeroes` or `ignore_zeros_in_jacobian` makes more sense to me...